text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,409 @@
+be small so this should be very fast.
+
+Further, this proposal suggests subdividing metadata images by project
+sub-team. Each sub-team will be given a unique reverse domain name like
+`io.paketo.java` and `io.paketo.utilities`. In this way, the project’s metadata
+can be easily combined without having conflicts.
+
+This allows users to pick and choose the metadata that’s relevant to their
+needs.
+
+## Buildpack Dependency Metadata Interface
+The interface between dependency metadata and a buildpack is simple. A single
+directory of metadata will be presented to the buildpack. It will be presented
+at the location specified by `BP_DEPENDENCY_METADATA`, which defaults to
+`/platform/deps/metadata` (the intent is to pilot and try out this or possibly
+other locations, eventually proposing an RFC with Cloud-Native buildpacks to
+standardize the location).
+
+Buildpacks do not care if there are multiple sources of metadata information,
+however, this needs to be merged and presented to the buildpack as one single
+directory. How that information is merged is outside of the scope of this
+document, but the directory structure defined in [Metadata
+Format](#metadata-format) guarantees that there will not be any duplicate
+dependency ids.
+
+This document does not specify how the dependency metadata folder should be
+provided to buildpacks but here are some possibilities:
+
+1. External. The metadata information can be managed outside of the buildpack
+   lifecycle. This allows for users to manually pull in the metadata they would
+   like when they would like it. The metadata can then be mapped into the build
+   container at the specified location using a volume mount to `pack build`.
+1. Via buildpack. We could add a Paketo buildpack to the beginning of buildpack
+   order groups that is responsible for pulling metadata and overrides
+   `BP_DEPENDENCY_METADATA` to point to its layer. The buildpack can check for
+   dependency metadata updates when it runs.
+1. Via builder. The builder could come with metadata included under
+   `/platform/deps/metadata`. This would likely require a builder with many
+   buildpacks to update very frequently though, so may not be the best idea for
+   the average case, however, it could be very useful as a way to distribute
+   dependencies in offline environments.
+1. Via the platform. Platforms may choose to offer enhanced functionality to
+   more easily distribute dependencies. In the end, the platform just needs to
+   ensure that the dependency metadata is available to the buildpack in the
+   required location.
+
+In addition to the flexibility of where the dependencies originate, this
+proposal also provides flexibility in how those dependencies are managed such
+as floating them so the latest versions are always available, pinning to a
+specific set of dependencies or even pinning and including the dependencies
+with the metadata.
```

Comment:
I think at a bare minimum in this RFC I would like to define some interface that would allow for the development of the decouple dependency metadata. I am not sure what exploration we would launch that would not be a competing interface to the one proposed. If you are looking for a competing interface or have one in mind then I think that we should have competing RFCs. I think that it would be fine to remove the potential delivery methods from this section but I think that the overall idea should remain.

Reply:
You might be right with the competing interface. I feel like it would make sense to go for `BP_DEPENDENCIES` and `/platform/deps` to allow dependency  metadata and assets to be distributed together (if assets get delivered together with metadata, i.e. offline/air gapped case).